### PR TITLE
server/swap: Do not increase reputation on self-match

### LINKED ISSUE #2751: rep: self-matches are earning a point
Self-matches are not supposed to give score, but they are.

### PR DESCRIPTION
**Pull request by martonp**
Closes #2751

### CODE REVIEW

File: server/swap/swap.go [hunk 1]
```diff
@@ -1832,7 +1832,7 @@ func (s *Swapper) processRedeem(msg *msgjson.Message, params *msgjson.Redeem, st
 	}
 
 	// Credit the user for completing the swap, adjusting the user's score.
-	if actor.user != counterParty.user || newStatus == order.MatchComplete { // if user is both sides, only credit on MatchComplete (taker done too)
```
Comment (JoeGruffins) [inline server/swap/swap.go:1835]: Should the `||` have been an `&&`? It looks fine with just the not counterparty check. User should get not credit for trading with themself.

**Review by JoeGruffins — APPROVED** [review]

File: server/swap/swap.go (hunk 1 above)
Comment (martonp) [reply server/swap/swap.go:1835]: I'm guessing the intention was to give someone one credit when trading with themselves instead of two.

Comment (buck54321) [reply server/swap/swap.go:1835]: Actually, you're both right. Someone did intend to give 1 point for setf-matches, but we shouldn't.

Comment (martonp) [reply server/swap/swap.go:1835]: Do we need the `newStatus == order.MatchComplete`? Why should the maker have to wait for the taker to also redeem before getting their score?

Comment (JoeGruffins) [reply server/swap/swap.go:1835]: No? Looks ok to me.

Comment (buck54321) [reply server/swap/swap.go:1835]: > Do we need the `newStatus == order.MatchComplete`? Why should the maker have to wait for the taker to also redeem before getting their score?

Oh right. No need for the `newStatus == order.MatchComplete`.

**Pull request merged**